skimage/transform/_warps.py

```diff
@@ -6,6 +6,14 @@
                          _convert_warp_input, _clip_warp_output)
 
 
+def _multichannel_default(multichannel, ndim):
+    # utility for maintaining previous color image default behavior
+    if ndim == 3:
+        return True
+    else:
+        return False
```

Comment:
We've discussed in the past that this is not a magic preferred by the team... But in this case it's different because you're not adding any magic, it was already there. However, the current selector is really imperfect because it will fail quite dramatically for 3D grayscale images. (Note also that you are ignoring the `multichannel` argument!!!)

I see two options for this:

1. If we want to maintain this magic, which to reiterate was already there:

```python
def _multichannel_default(multichannel, ndim, last_dim):
    if multichannel is not None:
        return multichannel
    if ndim == 3 and last_dim in [3, 4]:
        return True
    else:
        return False
```

2. if we want to deprecate this behaviour and specify a new default (i.e. universally `True` or `False`), we additionally need to raise a deprecation warning when `multichannel is None`.